test/integration/oracle/ChainlinkOEVMorphoWrapperIntegration.t.sol

```diff
@@ -0,0 +1,487 @@
+                    ts,
+                    uint80(1)
+                )
+            );
+            vm.expectRevert();
+            wrapper.latestRoundData();
+        }
+    }
+
+    function testUpdatePriceEarlyAndLiquidate_WELL() public {
+        _testLiquidation(
+            addresses.getAddress("CHAINLINK_WELL_USD_ORACLE_PROXY"),
+            addresses.getAddress("xWELL_PROXY"),
+            addresses.getAddress("MORPHO_CHAINLINK_WELL_USD_ORACLE"),
+            0.625e18,
+            1_000_000e18, // 1M WELL tokens (~$10k at $0.01)
+            10_000e18
+        );
+    }
+
+    function testUpdatePriceEarlyAndLiquidate_stkWELL() public {
+        // Note: stkWELL has no market, so fee recipient should be set to treasury
+        // The wrapper will automatically detect this and transfer instead of calling _addReserves
+        _testLiquidation(
+            addresses.getAddress("CHAINLINK_stkWELL_USD_ORACLE_PROXY"),
+            addresses.getAddress("STK_GOVTOKEN_PROXY"),
+            addresses.getAddress("MORPHO_CHAINLINK_stkWELL_USD_ORACLE"),
+            0.625e18,
+            1_000_000e18, // 1M stkWELL tokens
+            10_000e18
+        );
+    }
+
+    function testUpdatePriceEarlyAndLiquidate_MAMO() public {
+        _testLiquidation(
+            addresses.getAddress("CHAINLINK_MAMO_USD_ORACLE_PROXY"),
+            addresses.getAddress("MAMO"),
+            addresses.getAddress("MORPHO_CHAINLINK_MAMO_USD_ORACLE"),
+            0.385e18,
+            250_000e18, // Scale up collateral (5x from 50k to match WELL's economic value)
+            2_500e18 // Scale up seized amount proportionally (MAMO is 4x WELL price, so 10k/4 = 2.5k)
+        );
+    }
+
+    function _testLiquidation(
+        address wrapperAddr,
+        address collToken,
+        address oracleAddr,
+        uint256 lltv,
+        uint256 collateralAmount,
+        uint256 seized
+    ) internal {
+        ChainlinkOEVMorphoWrapper wrapper = ChainlinkOEVMorphoWrapper(
+            wrapperAddr
+        );
+        address loanToken = addresses.getAddress("USDC");
+        uint256 borrowAmount = 50e6; // $50 USDC (6 decimals)
+
+        // Setup market params
+        MarketParams memory params = MarketParams({
+            loanToken: loanToken,
+            collateralToken: collToken,
+            oracle: oracleAddr,
+            irm: addresses.getAddress("MORPHO_ADAPTIVE_CURVE_IRM"),
+            lltv: lltv
+        });
+
+        // Setup Morpho Blue
+        IMorphoBlue morpho = IMorphoBlue(addresses.getAddress("MORPHO_BLUE"));
+
+        // Setup borrower position
+        deal(collToken, BORROWER, collateralAmount);
+        vm.startPrank(BORROWER);
+        IERC20(collToken).approve(address(morpho), collateralAmount);
+        morpho.supplyCollateral(params, collateralAmount, BORROWER, "");
+
+        // Try to borrow - if it fails due to insufficient liquidity, skip the test
+        try morpho.borrow(params, borrowAmount, 0, BORROWER, BORROWER) {
```

Comment:
Would be great to mock liqudity on morpho so we always execute the test no matter the onchain state